agent/service.go

```diff
@@ -88,6 +88,12 @@ func New(ctx context.Context, logger *slog.Logger, eventSvc events.Service, cmp
 	svc.sm.StateFunctions[complete] = svc.publishEvent("in-progress", json.RawMessage{})
 	svc.sm.StateFunctions[running] = svc.runComputation
 
+	if len(cmp.Datasets) == 0 {
+		svc.sm.StateFunctions[receivingData] = func() {
+			svc.sm.SendEvent(dataReceived)
```

Comment:
Why are you sending `dataReceived` even when there was no data? This is a hack, and I wrote this before.

I do not like this - we should rather change the state machine not to even go in `receivingData` if there is no data, but to directly switch to `receivingAlgo` state